knime_extension/src/nodes/calculation.py

```diff
@@ -0,0 +1,494 @@
+# Convex Hull
+############################################
+@knext.node(
+    name="Convex Hull",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "ConvexHull.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Geo table",
+    description="Table with geometry column to transform",
+)
+@knext.output_table(
+    name="Transformed Geo table",
+    description="Transformed Geo input table",
+)
+@knut.geo_node_description(
+    short_description="This node generate the smallest convex Polygon containing all the points in each geometry.",
+    description="""This node generate the smallest convex Polygon containing all the points in each geometry.
+    The convex hull of a geometry is the smallest convex Polygon containing all the points in each geometry, 
+    unless the number of points in the geometric object is less than three. For two points, the convex hull 
+    collapses to a LineString; for 1, a Point.""",
+    references={
+        "Convex hull": "https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.convex_hull.html",
+    },
+)
+class ConvexHullNode(_SingleCalculator):
+
+    geo_col = knut.geo_col_parameter(
+        description="Select the geometry column to compute the convex hull."
+    )
+
+    def __init__(self):
+        super().__init__(lambda gdf: gdf.convex_hull, "hull", knut.TYPE_GEO)
+
+
+############################################
+# Coordinates node
+############################################
+@knext.node(
+    name="Coordinates XYZ",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "XYZcoordinates.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Geo table",
+    description="Table with a point geometry column to extract the coordinates from.",
+)
+@knext.output_table(
+    name="Table with coordinates",
+    description="Table with the coordinates for all point geometries",
+)
+@knut.geo_node_description(
+    short_description="Extracts the XYZ coordinates.",
+    description="This node extracts the XYZ coordinates for all given point objects."
+    + "The coordinates are appended to the input table as xyz coordinate columns.",
+    references={
+        "X coordinate": "https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.x.html",
+        "Y coordinate": "https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.y.html",
+        "Z coordinate": "https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.z.html",
+    },
+)
+class CoordinatesNode:
+
+    geo_col = knut.geo_point_col_parameter(
+        description="Select the point geometry column to extract the coordinates from."
+    )
+
+    add_z = knext.BoolParameter(
+        "Extract Z coordinate",
+        "Select this option to extract the Z coordinate. If not selected only the X and Y coordinate are extracted",
+        False,
+    )
+
+    col_x = "x"
```

Comment:
these are not used outside of the class, right? So they could have a `_` prefix to indicate that they are _private_

Reply:
Done